ui/src/views/Rules/ListRules/index.jsx

```diff
@@ -1074,8 +1073,28 @@ function ListRules(props) {
     return dialogStates[dialogState.mode];
   };
 
+  const filteredRulesToMatchSearchFilter = () => {
+    if (productChannelQueries && productChannelQueries[1]) {
+      const [product, channel] = productChannelQueries;
+      const filteredSet = filteredRulesWithScheduledChanges.filter(rule => {
+        return (
+          rule.product === product &&
+          (channel.includes(rule.channel) ||
+            channel.includes(
+              rule.channel.substring(0, rule.channel.length - 1)
```

Comment:
This is close, but simply comparing against every rule's channel-minus-the-last-character is not correct. In order for a rule to show up, one of the following must be true:
- `rule.channel` must match `channel` precisely
- `rule.channel` must end in a `*` AND the string before the `*` must be a prefix of `channel`.

(I realize this change works fine with all of the data currently in the system, but it may not always be correct.)